frontend/src/metabase/public/containers/PublicDashboard/PublicDashboard.tsx

```diff
@@ -90,16 +79,28 @@ const connector = connect(mapStateToProps, mapDispatchToProps);
 type ReduxProps = ConnectedProps<typeof connector>;
 
 type OwnProps = {
-  params: {
-    dashboardId?: DashboardId;
-    uuid?: string;
-    token?: string;
-  };
-} & DashboardControlsProps;
+  dashboardId: DashboardId;
+  queryParams: Record<string, string | string[] | null | undefined>;
```

Comment:
Just to confirm my understanding, I've seen `string | string[] | null | undefined` in some places and `ParameterValueOrArray | null` in others. I understand that the only difference is that query parameters are string only (or array of strings), while the parameter values could've been a number as well. Does that sound right?

```
export type ParameterValueOrArray = string | number | Array<any>;
```